### PR TITLE
Add BertMesh to models so that we can tag in BioASQ

### PR DESCRIPTION
**Pull request by nsorros**
## Description

Adds BertMesh in models by copying and not refactoring the experimental code in BertMesh. The reason being that at this point we want to maintain moving fast in experimentation while separating what we maintain for "production". As such we do not add code to train BertMesh as this happens still in experimentation mode. Once this becomes more stable we can do that.

DVC sort of up to date.

## Checklist

- [ ] Linked to Notion or GitHub issue
- [x] Added tests
- [ ] Updated README
- [x] DVC up to date

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ SCIENCE_TFIDF_SVM_MODEL:= tfidf-svm.pkl
 SCIENCE_SCIBERT_MODEL := scibert
 SCIENCE_LABEL_BINARIZER := label_binarizer.pkl
 
-PYTHON := python3.8
+PYTHON := python3.7
```
Comment (nsorros) [inline Makefile:15]: this goes back and forth according to the environment, maybe we should revert to python3 to not have to change often and ensure the code runs for both versions

Comment (aCampello) [reply Makefile:15]: Agree. or an env variable that the user can define? I keep changing to 3.8 as this is the one installed in my machine currently

File: grants_tagger/models/bert_mesh.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import scipy.sparse as sp
+import numpy as np
+import torch
+
+from grants_tagger.bertmesh.model import BertMesh
+
+
+class MeshDataset(Dataset):
+    def __init__(self, X):
+        self.tokenizer = BertTokenizerFast.from_pretrained("bert-base-uncased")
```
Comment (aCampello) [inline grants_tagger/models/bert_mesh.py:13]: Should the tokeniser size be a parameter?

I think it's called `max_position_embeddings`

File: grants_tagger/models/bert_mesh.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        self.model.eval()
+
+        Y_pred_proba = []
+        with torch.no_grad():
+            for inputs in tqdm(data):
+                outs = self.model(inputs)
+
+                Y_pred_proba_batch = outs.cpu().numpy()
+                Y_pred_proba_batch[Y_pred_proba_batch < self.cutoff_prob] = 0
```
Comment (aCampello) [inline grants_tagger/models/bert_mesh.py:62]: I don't understand why you do this.

File: grants_tagger/models/bert_mesh.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                Y_pred_proba.append(sp.csr_matrix(Y_pred_proba_batch))
```
Comment (aCampello) [inline grants_tagger/models/bert_mesh.py:63]: Ah, it's to use the sparse matrix trick. Nice

**Review by aCampello — APPROVED** [review]
The class is really nice, thanks for doing this. Can I be a massive pain and ask that you either add somewhere in the documentation the actual script/steps to tag (like the parameters passed to `python scripts/tag_bioasq_test_data` ? ) or add a dvc pipeline for that. Your choice.

**Comment by nsorros** [comment]
> The class is really nice, thanks for doing this. Can I be a massive pain and ask that you either add somewhere in the documentation the actual script/steps to tag (like the parameters passed to `python scripts/tag_bioasq_test_data` ? ) or add a dvc pipeline for that. Your choice.

Sure this is a good idea. Its kind of hacky at the moment.

File: grants_tagger/models/bert_mesh.py (hunk 4 above)
Comment (nsorros) [reply grants_tagger/models/bert_mesh.py:63]: Yep, we don't need all the probs, let's reduce the size.

File: grants_tagger/models/bert_mesh.py (hunk 2 above)
Comment (nsorros) [reply grants_tagger/models/bert_mesh.py:13]: what do you mean? I am not familiar with it

**Comment by nsorros** [comment]
This is ready to go. I will run the dvc pipeline on Monday to ensure it works smoothly in the latest batch and merge.

**Pull request merged**